test/gradcheck.jl

```diff
@@ -1553,46 +1553,13 @@ using Zygote: Buffer
   @test ∇W2 == W2
   @test ∇x == 6 .* x
 
-  @testset "incorrect promotion (#1352)" begin
-    u = [0.75, 0.5]
-    p = [-1.5, 0.05, 0.2, 0.01]
-
-    # in-place
-    function g1352!(du, u, p, t)
-      du[1, 1] = p[3] * u[1] + p[4] * u[2]
-      du[1, 2] = p[3] * u[1] + p[4] * u[2]
-      du[2, 1] = p[4] * u[1] + p[3] * u[2]
-      du[2, 2] = p[4] * u[1] + p[3] * u[2]
-      return nothing
-    end
-    du1_inplace, back_inplace = Zygote.pullback(u, p) do u, p
-      du = Zygote.Buffer(Matrix{Float64}(undef, 2, 2))
-      g1352!(du, u, p, 1.0)
-      return copy(du[:, 1])
-    end
-
-    # out-of-place
-    function g1352(u, p, t)
-      du11 = p[3] * u[1] + p[4] * u[2]
-      du12 = p[3] * u[1] + p[4] * u[2]
-      du21 = p[4] * u[1] + p[3] * u[2]
-      du22 = p[4] * u[1] + p[3] * u[2]
-      return [du11 du12
-              du21 du22]
-    end
-    du1, back = Zygote.pullback(u, p) do u, p
-      du = g1352(u, p, 1.0)
-      return du[:, 1]
-    end
+  # reduced mwe of #1352
+  @test Zygote.gradient([0,0]) do x
```

Comment:
Maybe use some more complicated function and some non-zero values? It happened to me before (eg in DistributionsAD) that too "nice" values and examples did not catch (all) problems.

Reply:
Sure, any suggestions? The test is mainly of the pullback of `getindex(x::Buffer, ::UnitRange)` which was not getting tested before (only `getindex(x::Buffer, ::Int)` was).

Reply:
Gentle ping on this. If there are no good ideas then I think this is ready?

Reply:
Fine by me to go ahead as-is or for someone to tweak the test. Just rebased on master for convenience.